Assets/Packages/com.studio23.ss2.splashscreensystem/Runtime/Data/ThirdPartyData.cs

```diff
@@ -0,0 +1,60 @@
+using Studio23.SS2.SplashScreenSystem.UI;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Studio23.SS2.SplashScreenSystem.Data
+{
+    [System.Serializable]
+    public class ThirdPartyData : SplashScreenData
+    {
+        public List<ThirdPartyDataEntry> ThirdPartyEntries = new List<ThirdPartyDataEntry>();
+
+        public override void UpdateAndShowSplashContainer(SplashScreenUIManager ui)
+        {
+            GridLayoutGroup gridLayout = ui.RectParent.GetComponent<GridLayoutGroup>();
+
+            foreach (var entry in ThirdPartyEntries)
+            {
+                GameObject newImage = Instantiate(ui.ImagePrefab, ui.RectParent);
+                Image newImageComponent = newImage.GetComponent<Image>();
+
+                newImageComponent.sprite = Sprite.Create(entry.image, new Rect(0, 0, entry.image.width, entry.image.height), Vector2.zero);
+                ui.ImageObjects.Add(newImage);
```

Comment:
If the Image component is all you need to fetch make necessary changes on, you can cache the prefab as an Image instead of GameObject. This will allow you to directly reference the Image component instead of having to use `GetComponent` for all entries.